nano/secure/ledger.cpp

```diff
@@ -563,8 +563,10 @@ void ledger_processor::receive_block (nano::receive_block & block_a)
 			result = block_a.valid_predecessor (*previous) ? nano::block_status::progress : nano::block_status::block_position;
 			if (result == nano::block_status::progress)
 			{
-				auto account (ledger.store.frontier.get (transaction, block_a.hashables.previous));
-				result = account.is_zero () ? nano::block_status::gap_previous : nano::block_status::progress; // Have we seen the previous block? No entries for account at all (Harmless)
+				auto account = ledger.account (*previous);
+				auto info = ledger.account_info (transaction, account);
+				debug_assert (info);
+				result = info->head != block_a.hashables.previous ? nano::block_status::fork : nano::block_status::progress; // Have we seen the previous block? No entries for account at all (Harmless)
```

Comment:
I see the handling of forks is fixed, but I believe the comment on this line should be updated to `// If we have the block but it's not the latest we have a signed fork `